src/tools/simple-kernel-timer-json/kp_kernel_info.h

```diff
@@ -6,7 +6,26 @@
 #include <sys/time.h>
 #include <cstring>
 
+#if defined(__GXX_ABI_VERSION)
+#define HAVE_GCC_ABI_DEMANGLE
+#endif
 
+#if defined(HAVE_GCC_ABI_DEMANGLE)
+#include <cxxabi.h>
+#endif // HAVE_GCC_ABI_DEMANGLE
+
+char* demangleName(char* kernelName)
+{
+#if defined(HAVE_GCC_ABI_DEMANGLE)
+	int status = -1;
+	char* demangledKernelName = abi::__cxa_demangle(kernelName, NULL, NULL, &status);
+	if (status==0) {
+		free(kernelName);
```

Comment:
This function almost certainly will leak memory, and it won't work if `kernelName` is a constant string like `"foo"`.  Why not just return a string?
```c++
char* demangledKernelName = abi::__cxa_demangle(kernelName, nullptr, nullptr, &status);
if (demangledKernelName != nullptr) {
  std::string name (demangledKernelName);
  free (demangledKernelName);
  return name;
}
else {
  return std::string (kernelName);
}
```